packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/ClassInfoReflectionModel.java

```diff
@@ -220,8 +222,25 @@ protected List<MethodInfoModel> prepareMethods() {
     }
 
     @Override
-    protected PackageInfoModel preparePackage() {
-        return PackageInfoModel.of(origin.getPackage());
+    protected List<PackageInfoModel> prepareAncestors() {
+        var classLoader = origin.getClassLoader();
+
+        return getAllAncestorPackageNames(origin.getPackageName())
+                .map(classLoader::getDefinedPackage).filter(Objects::nonNull)
+                .map(PackageInfoModel::of).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a stream of all ancestor package names, starting with the package
+     * itself.
+     *
+     * @param name
+     *            the package name
+     * @return the stream of all ancestor package names
+     */
+    private static Stream<String> getAllAncestorPackageNames(String name) {
```

Comment:
Maybe, drop the `Ancestor` part here?
```suggestion
    private static Stream<String> getAllPackageNames(String name) {
```

Reply:
As above, the purpose is to explain how multiple packages are related to a class name. I'd rather change the `name` parameter to `packageName`.

Reply:
And the "all" is because here we get them all, as names, even the ones that will be filtered out after.

Reply:
yeah, since we've agreed on the general naming, this one is no more necessary